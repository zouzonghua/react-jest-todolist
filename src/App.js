import React from 'react';

function App() {
  return (
    <div className="App" title="zonghua" data-test="container">
      hello world !
    </div>
  );
}

export default App;
