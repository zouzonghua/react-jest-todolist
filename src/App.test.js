import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from './App';

Enzyme.configure({ adapter: new Adapter() });

test('测试页面 App 组件', () => {
  // 单元测试使用 shallow
  // 集成测试使用 mount
  const wrapper = shallow(<App />)
  expect(wrapper).toMatchSnapshot()
  // const container = wrapper.find('[data-test="container"]')
  // console.log(wrapper.debug())
  // expect(container).toExist()
  // expect(container).toHaveProp('title', 'zonghua')
});
