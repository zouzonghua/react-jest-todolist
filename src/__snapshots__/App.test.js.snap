// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试页面 App 组件 1`] = `
<div
  className="App"
  data-test="container"
  title="zonghua"
>
  hello world !
</div>
`;
